# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1553106247 25200
#      Wed Mar 20 11:24:07 2019 -0700
# Node ID 6e01e5be8de335cd96f9a12521a4ffe1c9b3a78d
# Parent  ca8c2696af62d54a1e834a56f303cccd65dabb8f
methods.m: adapt method list post-processing to Java constructors (bug #55858)

* methods.m: Add regular expression to delete package prefix and argument list
from Java constructors.  Add regular expression to delete package prefix from
Java constructors when called with "-full".

diff --git a/scripts/miscellaneous/methods.m b/scripts/miscellaneous/methods.m
--- a/scripts/miscellaneous/methods.m
+++ b/scripts/miscellaneous/methods.m
@@ -72,17 +72,20 @@ function mtds = methods (obj, opt)
     mtds_list = mtds_list(:);  # return a column vector for compatibility
     havesigs = true;
   else
     error ("methods: invalid input argument");
   endif
 
   if (havesigs && ! showsigs)
     mtds_list = regexprep (mtds_list, '^(?:[^(]+) (\w+) ?\(.*$', '$1');
+    mtds_list = regexprep (mtds_list, '^(?:[\.\w]+\.)(\w+) ?\(.*$', '$1');
     mtds_list = unique (mtds_list);
+  else
+    mtds_list = regexprep (mtds_list, '^(?:[\.\w]+\.)(\w+ ?\(.*)$', '$1');
   endif
 
   if (nargout == 0)
     classname = ifelse (ischar (obj), obj, class (obj));
     printf ("Methods for class %s:\n", classname);
     disp (list_in_columns (mtds_list));
   else
     mtds = mtds_list;
